pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentDictionaryCreator.java

```diff
@@ -268,7 +268,7 @@ public int indexOfSV(Object value) {
         throw new UnsupportedOperationException("Unsupported data type : " + _fieldSpec.getDataType());
     }
 
-    Arrays.sort(indexes);
+//    Arrays.sort(indexes);
```

Comment:
Remove instead. Add comment in the PR description of this behavior change. Perhaps this should be done as a pre-requisite PR?

Reply:
Done.